### PR TITLE
Implemented decoding and hash rules for execution requests

### PR DESCRIPTION
**Pull request by lucassaldanha**
## PR Description
Implementing the rules for decoding and hashing execution requests according to https://github.com/ethereum/execution-apis/pull/591

## Fixed Issue(s)
part of #8620 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        Bytes.concatenate(
+            Bytes.of(ConsolidationRequest.REQUEST_TYPE),
+            consolidationRequestsSszList.sszSerialize()));
+  }
+
+  public Bytes32 hash(final ExecutionRequests executionRequests) {
+    final Bytes sortedEncodedRequests =
+        encode(executionRequests).stream()
+            // Encoded requests are sorted by their type (first byte)
+            .sorted(Comparator.comparingInt(b -> b.get(0)))
```
Comment (gfukushima) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java]: do we need to sort it here? haven't we already assembled a sorted list in encode()?

**Review by gfukushima — APPROVED** [review]
lgtm with a question

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java (hunk 1 above)
Comment (lucassaldanha) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java]: Yes, but it is a safeguard :) The performance impact is minimal given that it won't really need to re-order anything in the array.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ConsolidationRequest.java [hunk 2]
```diff
@@ -20,9 +20,14 @@
 import tech.pegasys.teku.infrastructure.ssz.tree.TreeNode;
 import tech.pegasys.teku.spec.datastructures.type.SszPublicKey;
 
+/*
+ https://eips.ethereum.org/EIPS/eip-7251
+*/
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ConsolidationRequest.java]: if we do need this comment at all, maybe single line?

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                      .asList());
+              break;
+            case ConsolidationRequest.REQUEST_TYPE:
+              // https://eips.ethereum.org/EIPS/eip-7251
+              executionRequestsBuilder.consolidations(
+                  executionRequestsSchema
+                      .getConsolidationRequestsSchema()
+                      .sszDeserialize(data.slice(1))
+                      .asList());
+              break;
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java]: nit: new syntax for case statements?

**Review by rolfyone — APPROVED** [review]
LGTM just nits

**Review by mkalinin — APPROVED** [review]
LGTM! 👍

**Review by StefanBratanov — APPROVED** [review]
LGTM. I think it makes sense to statically initialize these: `Bytes.of(DepositRequest.REQUEST_TYPE)` (maybe can turn those into enum not sure) to avoid creating all the time Bytes single byte instance.

**Pull request merged**